client/src/modules/roles/modal/userRole.js

```diff
@@ -0,0 +1,94 @@
+angular.module('bhima.controllers').controller('UsersRolesController', UsersRolesController);
+UsersRolesController.$inject = ['data', '$uibModal', '$rootScope',
+  '$uibModalStack', 'RolesService', 'SessionService', 'NotifyService'
+];
+
+function UsersRolesController(data, $uibModal, $rootScope, $uibModalStack, RolesService, session, Notify) {
+
+  var vm = this;
+  vm.close = close;
+  vm.user = angular.copy(data);
+  vm.loadRoles = loadRoles;
+  vm.getAffected = getAffected;
+  vm.roles = [];
+
+  // loa all roles
+  function loadRoles() {
+
+    RolesService.userRoles(vm.user.id, session.project.id)
+      .then(function (response) {
+        delete vm.gridOptions.data;
+        vm.roles = response.data;
+
+      })
+      .catch(function (err) {
+        console.log(err)
+      });
+  }
+
+  function getAffected() {
+    var codes = [];
+    var list = vm.roles;
+    for (var i = 0; i < list.length; i++) {
+      var role = list[i];
+      if (role.affected === 1) {
+        codes.push(role.uuid);
+      }
+
+    }
+    var data = {
+      user_id: vm.user.id,
+      role_uuids: codes
+    }
+    RolesService.affectToUser(data)
+      .then(function (response) {
+        Notify.success('FORM.INFO.OPERATION_SUCCESS');
+        vm.close();
+      }).catch(function (err) {
+        alert(erreur)
+      })
+
+  }
+  // ui-grid
+  var columns = [{
+      field: 'label',
+      displayName: 'Label',
+      exportable: 1,
+      exportWidth: 250
+    },
+    {
+      field: '-',
+      width: 100,
+      displayName: 'Affecté',
+      enableFiltering: false,
+      cellTemplate: `
```

Comment:
Please don't use template strings on the client.  They will break our production build.